.github/container/Dockerfile.t5x.arm64

```diff
@@ -0,0 +1,226 @@
+# build grain from a manually-patched version
+#------------------------------------------------------------------------------
+# TODO: Remove this once grain maintainers have published an arm64 wheel.
+
+FROM wheel-builder as grain-builder
+ARG PYTHON_VERSION="3.10"
+
+RUN <<"EOT" bash -exu
+set -o pipefail
+
+# Setup python
+apt-get update && apt-get install -y \
+    python3-dev python3-pip python3-venv && \
+    rm -rf /var/lib/apt/lists/* && \
+    python${PYTHON_VERSION} -m pip install pip --upgrade && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python${PYTHON_VERSION} 0
+
+# NOTE: We are ignoring the fact that "array_record" will actually resolve to the wrong version
+#   here. When we put it all together in the final image, we make sure that the right version is
+#   already installed.
+pip install \
+    absl-py \
+    array_record \
+    build \
+    cloudpickle \
+    dm-tree \
+    etils[epath] \
+    "more-itertools>=9.1.0" \
+    numpy;
+
+pip install \
+    dill \
+    jax \
+    jaxlib \
+    tensorflow \
+    tensorflow-datasets;
+
+git clone https://github.com/gspschmid/grain /opt/grain
+cd /opt/grain
+git checkout external-arm64-build-poc
+
+chmod a+x ./grain/oss/build_whl.sh
+./grain/oss/build_whl.sh
+
+ls /tmp/grain/all_dist/*.whl
+EOT
+
+
+###############################################################################
+## T5X for AArch64
+###############################################################################
+
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE} as mealkit
+ARG REPO_T5X
+ARG REF_T5X
+ARG SRC_PATH_T5X
+
+COPY --from=array_record-builder /tmp/array_record/all_dist/array_record*linux_aarch64.whl /opt/
+RUN echo "array_record @ file://$(ls /opt/array_record*.whl)" >> /opt/pip-tools.d/manifest.pax
+
+COPY --from=grain-builder /tmp/grain/all_dist/grain*.whl /opt/
+RUN echo "grain @ file://$(ls /opt/grain*.whl)" >> /opt/pip-tools.d/manifest.pax
+
+
+### --- TO BE REFACTORED? (everything below is as in @nouiz's original PR) ---
+
+# force a recent version to have latest protobuf dep
+# Install now as some build need them.
+RUN pip install "tensorflow_datasets==4.9.2"
+RUN pip install "tensorflow==2.13.0"
+RUN pip install "chex==0.1.7"
+RUN <<"EOF" bash -ex
+cd /opt
+git clone http://github.com/tensorflow/text.git
+cd text
+git checkout v2.13.0
+EOF
+
+RUN <<"EOF" bash -ex
+cd /opt/text
+
+wget https://github.com/bazelbuild/bazelisk/releases/download/v1.17.0/bazelisk-linux-arm64 -O /usr/bin/bazel ;
+chmod a+x /usr/bin/bazel
+
+./oss_scripts/run_build.sh
+find * | grep '.whl$'
+EOF
+RUN pip install /opt/text/tensorflow_text-*.whl
+#RUN echo "-e file:///opt/text/tensorflow_text-*.whl"  >> /opt/pip-tools.d/manifest.t5x
+RUN echo "tensorflow-text @  file://$(ls /opt/text/tensorflow_text-*.whl)" >> /opt/pip-tools.d/manifest.t5x
+
+# Install T5 now, Pip will build the wheel from source, it needs Rust.
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > /tmp/rustup.sh && \
```

Comment:
Could you please validate if Rust is still needed as of now? Can we install that  in a `builder` stage too and only copy out the wheel?

Reply:
Seems to work without -- which is nice because we can elide the explicit `pip install t5` entirely now.